src/ucp/api/ucp.h

```diff
@@ -242,9 +242,29 @@ enum ucp_ep_close_mode {
                                               for all endpoints created on
                                               both (local and remote) sides to
                                               avoid undefined behavior. */
-    UCP_EP_CLOSE_MODE_FLUSH         = 1  /**< @ref ucp_ep_close_nb schedules
+    UCP_EP_CLOSE_MODE_FLUSH         = 1, /**< @ref ucp_ep_close_nb schedules
                                               flushes on all outstanding
                                               operations. */
+    UCP_EP_CLOSE_MODE_SYNC          = 2  /**< <ul>
+                                              <li>If there is a connected
+                                              remote endpoint, initiate a
+                                              synchronization with the peer, and
+                                              the request returned from @ref
+                                              ucp_ep_close_nb will be completed
+                                              after the local endpoint is
+                                              flushed and the remote endpoint is
+                                              disconnected. @note The remote
```

Comment:
I'm not sure if I understand the semantics defined here. Adding @gmegan.
Does the destroy the endpoint only when the remote endpoint is destroyed ?

Reply:
I would say as it's described - the destroy the endpoint only when the remote endpoint is disconnected. In your wording it looks like deadlock.

Reply:
@shamisp the endpoint handle can no longer be used locally after this function is called.
in addition, if the endpoint is connected to a remote endpoint, it would also generate a "close" event on the remote peer and put the remote endpoint into error state.

Reply:
It isn't  worded very clear. @gmegan or @bbenton to you want to give it a try ?

Reply:
/** @ref ucp_ep_close_nb will flush the local endpoint and handle any existing remote endpoint by generating a close event on the remote peer and putting the remote endpoint into an error state.
<ul> 
 <li>If there is a connected remote endpoint, initiate a synchronization with the peer. The remote
                                              endpoint will get a notification
                                              through the callback defined in
                                              ucp_ep_params_t::err_handler with
                                              @ref UCS_ERR_REMOTE_DISCONNECT
                                              status, unless it also has closed
                                              its endpoint. </li>
                                              <li> If the endpoint is not
                                              connected to a remote endpoint,
                                              the behavior will be the same as
                                              @ref UCP_EP_CLOSE_MODE_FLUSH.</li>
                                              </ul> */

Reply:
Thanks! Updated.

Reply:
I like the new working. THanks !